docs/src/references/c8y-configuration-management.md

```diff
@@ -162,4 +162,240 @@ Note that:
+        Tedge Agent->>C8Y Cfg Plugin: 4: HTTP Response, contains tmp-URL, used to access uploaded file
+
+        C8Y Cfg Plugin->>External Device: 5: MQTT notification for "child1": Requst to download config type "bar1" on <tmp-url>
+
+        External Device->>Tedge Agent: 6: HTTP GET <tmp-url>
+        External Device->>External Device: 7: Apply downloaded file as config type "bar1"
+        External Device->>C8Y Cfg Plugin: 8: MQTT Notification: config type "bar1" applied successully
+```
+
+  1) At some point a ConfigDownloadRequest sent from cloud for type `bar.conf` for `child1` arrives at C8Y config plugin.
+     The format of C8Y SmartREST message for config send operation is: `524,<childid>,<URL>,<config type>`. See [C8Y SmartREST doc](https://cumulocity.com/guides/reference/smartrest-two/#download-configuration-file-with-type-524)<br/>
+    Example: `524,child1,http://www.my.url,bar.conf`
+  2) C8Y config plugin: downloads the file based on the URL received from C8Y, to some temporary location
+
+  3) C8Y config plugin: uploads the file to the thin-edge filetransfer feature, and removes it from the temporary location
+  
+    HTTP PUT request: `http://<ip address of thin-edge devicee>/tedge/tmpfiles`
+    
+  4) The response from the HTTP filetransfer feature contains a _temporary URL_, that is later used to uniquely access that uploaded file.<br/>
+    Example for a _temporary URL_: `/tedge/tmpfiles/<random string>`
+
+  5) C8Y config plugin: notifies the external device `child1` via MQTT to download the new configuration from the thin-edge device with the _temporary URL_.
+
+    Topic: `tedge/configuration/req/send/{config type}/{childid}`<br/>
+    Example: `tedge/configuration/req/send/bar.conf/child1`<br/>  
+    Payload: ` <temporary URL> `<br/>
+    TODO: Investigate and decide about topic structure and payload in scheduled prototype (https://github.com/thin-edge/thin-edge.io/issues/1307) 
+
+  6) external device child1: Downloads configuration file from the thin-edge device with the HTTP filetransfer feature and the _temporary URL_, and filetransfer removes the file from it's local filesystem folder.
+  
+    Example for HTTP GET request: `http://<ip address of thin-edge device>/<temporary URL>`
+    
+  7) external device child1: applies the new configuration and notifies the the plugin via MQTT about success.
+ 
+    Topic: `tedge/configuration/res/send/{config type}/{childid}`<br/>
+    Example: `tedge/configuration/res/send/bar.conf/child1`<br/>
+       
+    TODO: Investigate and decide about topic structure and payload in scheduled prototype (https://github.com/thin-edge/thin-edge.io/issues/1307). Unhappy paths also to be considered here.
+
+  8) C8Y config plugin: recognizes the MQTT notification and sends the result to C8Y.
+
+    TO-BE-DEFINED:
+    
+      * Decide which error paths to be considered.
+      * Decide if any timeout to be considerd when waiting for upload notification from external device.
+
```

Comment:
It might be worth highlighting the fact that the supported config list updates are also pushed from the cloud to the child device using this same mechanism where a an updated `c8y-config-plugin.toml` is pushed to the device with the type `c8y-config-plugin`, and the child device is expected to update its own config list with the contents of this file.

Reply:
Just to be clear: Are you talking about the per child file `/etc/tedge/c8y/<childid>/c8y-configuration-plugin.toml`, that is implicitely added to each child-device's config list? And especially about the fact that in case of any change of that file (whether from cloud or a local process) an update request is sent to the external device (similar as for all other config types)?

Reply:
Exactly.